File: MessageManager/MessageManager/Models/Message.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text.Json.Serialization;
using BibleReferenceParser.Parsing;
using MessageManager.Utility;

namespace MessageManager.Models
{
    public class Message
    {
        public int Id { get; set; }

        [MinLength(3)]
        [StringLength(120)]
        [RegularExpression("^[A-Z0-9].*$")]
        [Required]
        public string Title { get; set; }

        [StringLength(1024)]
        public string Description { get; set; }


        [DataType(DataType.Date)]
        [Required]
        public DateTime Date { get; set; }

        public int? VideoId { get; set; }
        public Video Video { get; set; }

        public int? AudioId { get; set; }
        public Audio Audio { get; set; }

        public int? NotesId { get; set; }
        public Notes Notes { get; set; }

        public int? SeriesId { get; set; }

        [JsonIgnore]
        public Series Series { get; set; }

        [Display(Name = "Bible References")]
        public List<BibleReferenceRange> BibleReferences { get; set; }

        [NotMapped]
        [Display(Name = "Bible References")]
        [BibleReferenceValidation]
        public string BibleReferencesString
        {
            get
            {
                if (BibleReferences == null)
                {
                    return "";
                }
                var bibleReferenceStrings = BibleReferences.Select(x => x.ToFriendlyString());
                return string.Join(", ", bibleReferenceStrings);
            }
            set
            {
                var convertedList = new List<BibleReferenceRange>();
                if (value != null)
                {
                    var parsedList = Parser.Parse(value);
                    foreach (var referenceRange in parsedList)
                    {
                        var referenceRangeModel = BibleReferenceRange.From(referenceRange);
                        referenceRangeModel.MessageId = Id;
                        convertedList.Add(referenceRangeModel);
                    }
                }
                BibleReferences = convertedList;
            }
        }

        public override string ToString()
        {
            return $"Message(Id={Id}, " +
                   $"Title={Title}, " +
                   $"Description={Description}, " +
                   $"AudioId={AudioId}, " +
                   $"NotesId={NotesId}, " +
                   $"VideoId={VideoId}, " +
                   $"SeriesId={SeriesId})";
        }
    }
}

File: message-manager/backend/src/models/message/Message.js
const Sequelize = require('sequelize');
module.exports = function(sequelize, DataTypes) {
  return sequelize.define('Message', {
    id: {
      autoIncrement: true,
      type: DataTypes.INTEGER,
      allowNull: false,
      primaryKey: true
    },
    title: {
      type: DataTypes.STRING(120),
      allowNull: false
    },
    description: {
      type: DataTypes.STRING(1024),
      allowNull: true
    },
    date: {
      type: DataTypes.DATE,
      allowNull: false
    },
    videoId: {
      type: DataTypes.INTEGER,
      allowNull: true
    },
    audioId: {
      type: DataTypes.INTEGER,
      allowNull: true
    },
    notesId: {
      type: DataTypes.INTEGER,
      allowNull: true
    },
    seriesId: {
      type: DataTypes.INTEGER,
      allowNull: true,
      references: {
        model: 'Series',
        key: 'id'
      }
    }
  }, {
    sequelize,
    tableName: 'Message',
    schema: 'dbo',
    timestamps: false,
    indexes: [
      {
        name: "IX_Message_SeriesId",
        fields: [
          { name: "seriesId" },
        ]
      },
      {
        name: "PK_Message",
        unique: true,
        fields: [
          { name: "id" },
        ]
      },
    ]
  });
};
